plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java

```diff
@@ -64,23 +79,27 @@ public ServletClasspathProvider() {
   // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/953
   @Override
   public IClasspathEntry[] resolveClasspathContainer(IRuntime runtime) {
-    return doResolveClasspathContainer();
+    return doResolveClasspathContainer(DEFAULT_DYNAMIC_WEB_VERSION);
   }
 
-  private IClasspathEntry[] doResolveClasspathContainer() {
+  private IClasspathEntry[] doResolveClasspathContainer(IProjectFacetVersion dynamicWebVersion) {
+
+    String servletApiId;
+    String jspApiId;
+    if (WebFacetUtils.WEB_31.equals(dynamicWebVersion)
+        || WebFacetUtils.WEB_30.equals(dynamicWebVersion)) {
+      servletApiId = "servlet-api-3.1";
```

Comment:
We could set up a `<library>` clause with the Servlet 3.0 and JSP 2._whatever_, but since 3.1 is backward compatible it didn't seem worth the while?

If we were going to change this, I think it would be better to create `<library>` entries for the specific supported `jst.web` versions.